types/router.d.ts

```diff
@@ -89,7 +89,7 @@ export interface PathToRegexpOptions {
   end?: boolean
 }
 
-export interface RouteConfig {
+interface BaseRouteConfig {
```

Comment:
Let's change the name

```suggestion
interface _RouteConfigBase {
```